src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,80 +1,59 @@
 package core.basesyntax;
 
-import java.io.BufferedReader;
-import java.io.BufferedWriter;
 import java.io.File;
-import java.io.FileReader;
-import java.io.FileWriter;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.List;
 
 public class WorkWithFile {
-    private static final int INDEX_OF_SUPPLY = 0;
-    private static final int INDEX_OF_BUY = 1;
-    private static final int INDEX_OF_RESULT = 2;
+
+    private static final String COMA = ",";
     private static final int INDEX_OF_SUPPLY_OR_BUY = 0;
     private static final int INDEX_OF_QUANTITY = 1;
-    private static final int REQUIRED_CAPACITY = 3;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private int buyValue = 0;
+    private int supplyValue = 0;
 
     public void getStatistic(String fromFileName, String toFileName) {
-        int[] supplyAndBuyAndResult = getSupplyAndBuyAndResult(fromFileName);
-        createReport(supplyAndBuyAndResult, toFileName);
+        readFromFile(fromFileName);
+        StringBuilder report = createReport();
+        writeToFile(report, toFileName);
     }
 
-    private int[] getSupplyAndBuyAndResult(String fromFileName) {
+    private void readFromFile(String fromFileName) {
         File file = new File(fromFileName);
-        int[] supplyAndBuyAndResult = new int[REQUIRED_CAPACITY];
-        String[] split;
+        String[] splitLine;
         try {
-            BufferedReader reader = new BufferedReader(new FileReader(file));
-            String lineOfFile = reader.readLine();
-
-            while (lineOfFile != null) {
-                split = lineOfFile.split(",");
-                if (split[INDEX_OF_SUPPLY_OR_BUY].equals("supply")) {
-                    supplyAndBuyAndResult[INDEX_OF_SUPPLY]
-                            += Integer.parseInt(split[INDEX_OF_QUANTITY]);
-                } else if (split[INDEX_OF_SUPPLY_OR_BUY].equals("buy")) {
-                    supplyAndBuyAndResult[INDEX_OF_BUY]
-                            += Integer.parseInt(split[INDEX_OF_QUANTITY]);
+            List<String> allLines = Files.readAllLines(file.toPath());
+            for (String allLine : allLines) {
+                splitLine = allLine.split(COMA);
+                if (splitLine[INDEX_OF_SUPPLY_OR_BUY].equals(SUPPLY)) {
+                    supplyValue = supplyValue + Integer.parseInt(splitLine[INDEX_OF_QUANTITY]);
+                } else if (splitLine[INDEX_OF_SUPPLY_OR_BUY].equals(BUY)) {
+                    buyValue = buyValue + Integer.parseInt(splitLine[INDEX_OF_QUANTITY]);
                 }
```

Comment:
lets also move the counting logic into a separate method for `readFromFile` must be responsible for reading the file only.

Reply:
done